packages/graphql/lib/src/core/query_options.dart

```diff
@@ -33,6 +40,9 @@ class QueryOptions<TParsed extends Object?> extends BaseOptions<TParsed> {
           parserFn: parserFn,
         );
 
+  final OnQueryCompleted? onCompleted;
+  final OnQueryError? onError;
```

Comment:
You'll need to add these two to the `properties` getter below. This allows us to correctly compute hash and equality between two options.